### PR TITLE
chore: add documentation on custom destination caching

### PR DESCRIPTION
**Pull request by deekshas8**
## What Has Changed?

This PR adds documentation on customizable destination caching. 

## Manual Checks?

- [ ] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [ ] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [ ] You formatted all changed files with prettier (`npm run prettier`)
- [ ] You tested if the documentation still builds (`npm run build`)
- [ ] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [ ] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/js/features/connectivity/destination-cache-isolation.mdx [hunk 1]
```diff
@@ -112,6 +115,38 @@ You need to set it manually.
 You need to activate caching explicitly with the `useCache` option, even if you pass an isolation strategy.
 :::
 
+### Custom Destination-Service Cache
+
+When the cache option is enabled, the SAP Cloud SDK stores the fetched destinations in an in-memory cache.
+However, in certain cases, a persistent or distributed cache may be more suitable according to your application architecture.
+For such cases where a different cache storage is desired, the SAP Cloud SDK allows you to plug-in your own implementation for destination cache.
+
+To create your custom destination cache provider, implement the `DestinationCacheInterface` interface and define the `get`, `set` and `clear` methods.
```
Comment (FrankEssenberger) [inline docs/js/features/connectivity/destination-cache-isolation.mdx:124]: [nth] add a link to the API doc - I guess this is only possible after publish. Perhaps make a note to add this.

File: docs/js/features/connectivity/destination-cache-isolation.mdx [hunk 2]
```diff
@@ -112,6 +115,38 @@ You need to set it manually.
...
+
+Once you have implemented your custom cache provider, you can tell the SAP Cloud SDK to use it in place of the default one.
+To do this, call the method `setDestinationCache` at the start of your application code, i.e., before the first calls to either the `getDestination` or the generic client's `executeHttpRequest` methods.  
+Pass your cache provider instance as an argument to this method.
+
+```ts
+setDestinationCache(new CustomCacheProvider({ expirationTime }));
+```
+
+All subsequent calls to fetch destinations will use this cache.
```
Comment (FrankEssenberger) [inline docs/js/features/connectivity/destination-cache-isolation.mdx:148]:
```suggestion
All subsequent calls to fetch destinations will use the custom cache.
```

**Review by FrankEssenberger — APPROVED** [review]
LGTM - only small questions and suggestions

File: docs/js/features/connectivity/destination-cache-isolation.mdx [hunk 3]
```diff
@@ -112,6 +115,38 @@ You need to set it manually.
...
+
+- `entry`, the item you want to store, and
+- `expires`, the expiration time for an individual cache item. This is optional.
+
+Once you have implemented your custom cache provider, you can tell the SAP Cloud SDK to use it in place of the default one.
+To do this, call the method `setDestinationCache` at the start of your application code, i.e., before the first calls to either the `getDestination` or the generic client's `executeHttpRequest` methods.  
+Pass your cache provider instance as an argument to this method.
+
+```ts
+setDestinationCache(new CustomCacheProvider({ expirationTime }));
```
[the remark this answers is not in the document]
Comment (deekshas8) [reply docs/js/features/connectivity/destination-cache-isolation.mdx:145]: No, it's not enforced. I'll update the example for clarity

**Pull request merged**